pygmt/datatypes/dataset.py

```diff
@@ -211,5 +213,5 @@ def to_dataframe(self) -> pd.DataFrame:
                 pd.Series(data=np.char.decode(textvector), dtype=pd.StringDtype())
             )
 
-        df = pd.concat(objs=vectors, axis=1)
+        df = pd.concat(objs=vectors, axis=1) if vectors else pd.DataFrame()
```

Comment:
The empty `pd.DataFrame()` won't have any columns. Should there still be columns returned (even if there are no rows)? How would this work with #3117 for example?

Reply:
I guess it's fine.
```
In [1]: import pandas as pd

In [2]: df = pd.DataFrame(columns=None)

In [3]: df.empty
Out[3]: True

In [4]: df = pd.DataFrame(columns=None)

In [5]: df = pd.DataFrame(columns=["col1", "col2"])

In [6]: df.empty
Out[6]: True
```

Reply:
Column names are set here like so:

https://github.com/GenericMappingTools/pygmt/blob/1eb6dec5d72a29bc00349459d1369eb0a0673a93/pygmt/clib/session.py#L1853-L1861

So we would do something like:

```python
import pandas as pd

df = pd.DataFrame(columns=None)
assert df.empty
df.columns = ["col1", "col2"]
```

But setting column names to `["col1", "col2"]` errors with:

```python-traceback
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[8], line 1
----> 1 df.columns = ["col1", "col2"]

File ~/mambaforge/envs/pygmt/lib/python3.12/site-packages/pandas/core/generic.py:6310, in NDFrame.__setattr__(self, name, value)
   6308 try:
   6309     object.__getattribute__(self, name)
-> 6310     return object.__setattr__(self, name, value)
   6311 except AttributeError:
   6312     pass

File properties.pyx:69, in pandas._libs.properties.AxisProperty.__set__()

File ~/mambaforge/envs/pygmt/lib/python3.12/site-packages/pandas/core/generic.py:813, in NDFrame._set_axis(self, axis, labels)
    808 """
    809 This is called from the cython code when we set the `index` attribute
    810 directly, e.g. `series.index = [1, 2, 3]`.
    811 """
    812 labels = ensure_index(labels)
--> 813 self._mgr.set_axis(axis, labels)
    814 self._clear_item_cache()

File ~/mambaforge/envs/pygmt/lib/python3.12/site-packages/pandas/core/internals/managers.py:238, in BaseBlockManager.set_axis(self, axis, new_labels)
    236 def set_axis(self, axis: AxisInt, new_labels: Index) -> None:
    237     # Caller is responsible for ensuring we have an Index object.
--> 238     self._validate_set_axis(axis, new_labels)
    239     self.axes[axis] = new_labels

File ~/mambaforge/envs/pygmt/lib/python3.12/site-packages/pandas/core/internals/base.py:98, in DataManager._validate_set_axis(self, axis, new_labels)
     95     pass
     97 elif new_len != old_len:
---> 98     raise ValueError(
     99         f"Length mismatch: Expected axis has {old_len} elements, new "
    100         f"values have {new_len} elements"
    101     )

ValueError: Length mismatch: Expected axis has 0 elements, new values have 2 elements
```

Reply:
I think we need to refactor `GMT_DATASET.to_dataframe()` to accept more panda-specific parameters (e.g., `column_names`, `index_col`). The the `virtualfile_to_dataset` will be called like:
```
 result = self.read_virtualfile(vfname, kind="dataset").contents.to_dataframe(columns=column_names) 
 if output_type == "numpy":  # numpy.ndarray output 
     return result.to_numpy() 
  
 return result  # pandas.DataFrame output
```

Reply:
Right, so moving these lines that handle the column names from `virtualfile_to_dataset` to `to_dataframe`:

https://github.com/GenericMappingTools/pygmt/blob/62eb5d6b9156612c89d4d0357904fd09e2dddc17/pygmt/clib/session.py#L1861-L1863

Do you want to just do that in #3117? Or have a separate PR to handle this?

Reply:
Better to do it in a separate PR so that #3117 can focus on parsing the column names from header.

Reply:
> Right, so moving these lines that handle the column names from `virtualfile_to_dataset` to `to_dataframe`:
> 
> https://github.com/GenericMappingTools/pygmt/blob/62eb5d6b9156612c89d4d0357904fd09e2dddc17/pygmt/clib/session.py#L1861-L1863
> 
> Do you want to just do that in #3117? Or have a separate PR to handle this?

Done in #3140, so need to refactor this PR after #3140 is merged.